week1/Q1_test.py

```diff
@@ -0,0 +1,19 @@
+import unittest
+import Q1
+
+
+class PermutationTest(unittest.TestCase):
+
+	def testExampleCases(self):
```

Comment:
Empty string is probably the most important.

In addition, you can check things explicitly to represent some of the things that are important for the function to work correctly (which might be already coverage by some of the examples). For example:
- make sure case does not matter:
  check "abc" against "ABC"
- make sure number of occurrences does not matter:
  check "aab" matches "baa" but not "ba" or "baaa"